### PR TITLE
Base Node CLI startup fixes

### PR DESCRIPTION
**Pull request by mikethetike**
* Add return exit codes
* Set the default logger to output to console from info up
* Added CLI args to prompt the user to create default config data if it does not exist
* Fixed some of the default config values which were incorrect

## Description
The main aim of this PR is to smooth out the first run experience for the CLI base node. Hope it helps. 

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: common/logging/log4rs-sample.yml [hunk 1]
```diff
@@ -32,9 +32,10 @@ appenders:
 
 # Set the default logging level to "error" and attach the "stdout" appender to the root
```
Comment (philipr-za) [inline common/logging/log4rs-sample.yml]: Maybe sync up the comment with the code

File: common/src/configuration.rs [hunk 2]
```diff
@@ -311,7 +311,7 @@ pub fn default_config() -> Config {
         default_subdir("mainnet/node_id.json"),
     )
     .unwrap();
-    cfg.set_default("base_node.mainnet.address", "http://localhost:18089")
+    cfg.set_default("base_node.mainnet.address", "/ip4/0.0.0.0/tcp/18089/http")
```
Comment (CjS77) [inline common/src/configuration.rs]: 127.0.0.1 ?

File: common/logging/log4rs-sample.yml [hunk 3]
```diff
@@ -32,9 +32,10 @@ appenders:
...
 root:
-  level: warn
+  level: info
   appenders:
     - base_layer
+    - stdout
```
Comment (CjS77) [inline common/logging/log4rs-sample.yml]: We've got a debug sample logger for this.

The production logger is intentionally keeping stdout clean.
You should be using a log util like lnav on the log files themselves to read logs.

File: common/src/configuration.rs (hunk 2 above)
Comment (mikethetike) [reply common/src/configuration.rs]: Could be, but I think that will prevent it from accepting external inbound connections

File: common/logging/log4rs-sample.yml (hunk 3 above)
Comment (mikethetike) [reply common/logging/log4rs-sample.yml]: The problem is actually that without this, errors are not finding their way to stdout. The program starts logging to stdout, but then after the config is loaded, redirects all logs to the file instead and then ends.
```
$ Tari Base Node
$ Copyright 2019-2020. The Tari Community
$ Version 0.0.5

Press Ctrl-C to quit..
Config file does not exist. Would you like to create a new one (Y/n)?

Installing new config file at /Users/mikethetike/.tari/config.toml
Logging configuration file does not exist. Would you like to create a new one (Y/n)?

Installing new logfile configuration at /Users/mikethetike/.tari/log4rs.yml
Initializing logging according to "/Users/mikethetike/.tari/log4rs.yml"
<program ends>
$ 
```
Took me a while to realise that the error was going to the file.

Comment (CjS77) [reply common/logging/log4rs-sample.yml]: Ok, but then let's fix the config file to do what we want.

The logger must be called `base_node` and not `stdout`. Then it works:

```
$ Tari Base Node
$ Copyright 2019-2020. The Tari Community
$ Version 0.0.5

Press Ctrl-C to quit..
Initializing logging according to "/Users/cayle/.tari/log4rs.yml"
ERROR Node identity information not found. The node identity file, /Users/cayle/.tari/testnet/node_id.json, has an error. j s o n error. You can update the configuration file to point to a valid node identity file, or re-run the node with the --create_id flag to create anew identity.
```

File: common/src/configuration.rs (hunk 2 above)
Comment (CjS77) [reply common/src/configuration.rs]: I thought these were tests for some reason. Yup, that's fine.

File: common/logging/log4rs-sample.yml (hunk 3 above)
Comment (mikethetike) [reply common/logging/log4rs-sample.yml]: Done

**Pull request merged**